### PR TITLE
Back button in Navigation should go linearly back

### PR DESCRIPTION
**Pull request by gadabra**
I try to solve sixth item in this https://github.com/saving-satoshi/saving-satoshi/issues/66#issuecomment-1435354590

I create `useLessonNav.tsx` to handle url routing logic in a separate file and refactor navigation in NavBar component

### CODE REVIEW

**Comment by gadabra** [comment]
I was misinterpret #155 .

This https://github.com/saving-satoshi/saving-satoshi/issues/155#issuecomment-1442057096  must be here. I think abstracting `Next` lesson path is good idea.  @afterburn, what do you think? Should I implement under this pr?

File: hooks/useLessonNav.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    currentChapterAndLesson = chapterId
+  } else {
+    currentChapterAndLesson =
+      chapterId + '/' + lessons[indexOfCurrentLesson - 1]
+  }
+
+  return `${lang}/chapters/${currentChapterAndLesson}`
+}
+
+export const useLessonNav = () => {
```
Comment (afterburn) [inline hooks/useLessonNav.ts:42]: Let's rename this to `useContentRoute`, this way the name will still make sense if/when we add more properties to the return value (e.g. `chapterPath`).

File: hooks/useLessonNav.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { usePathname } from 'next/navigation'
+import { chapters } from 'content'
+
+const handlePath = (pathName: string) => {
```
Comment (afterburn) [inline hooks/useLessonNav.ts:4]: Rename this to `getPathData` or something similar, `handlePath` does not imply it returns some data about the path.

**Comment by afterburn** [comment]
> I was misinterpret #155 .
> 
> This [#155 (comment)](https://github.com/saving-satoshi/saving-satoshi/issues/155#issuecomment-1442057096) must be here. I think abstracting `Next` lesson path is good idea. @afterburn, what do you think? Should I implement under this pr?

Not sure I understand what you mean.. In this PR you added a hook that lets you get the path of the previous lesson, you are suggesting to expand it so it also returns the next lesson path? If that is what you mean, then yes, I think thats a good idea! 😁

**Comment by gadabra** [comment]
@afterburn  Yes, exactly.

**Comment by afterburn** [comment]
> @afterburn Yes, exactly.

Okay good, I like the idea. Feel free to add the `nextLessonPath` to the return value of the `useContentRoute` hook in this PR, and request a review from me when its ready! :grin:

**Comment by afterburn** [comment]
@gadabra I just noticed @GBKS [mentioned something](https://github.com/saving-satoshi/saving-satoshi/issues/66#issuecomment-1448280976) about the behavior you implemented in this PR. Just to be clear, no matter the outcome of that discussion, I think the new hook you added (`useContentRoute`) is still very helpful for us, so please leave that in.

**Comment by ecurrencyhodler** [comment]
Hey Gadabra. 

Christoph is going to redesign a new icon and that will redirect to another page on the site. 

I'm going to create a separate issue for this. I think we should strip the PR except for the code kevin highlighted and create a new PR for the new issue.

What do you think?

**Comment by afterburn** [comment]
Hey @gadabra, if you have any questions or if something is not clear please don't hesitate to reach out to me, @benalleng or @ecurrencyhodler!

**Comment by ecurrencyhodler** [comment]
Okay. Let's close this PR then and you can submit another PR. Let me know if you need me to reopen it. Thanks.

And I agree. Let's not let the icon be a blocker. Go ahead and address https://github.com/saving-satoshi/saving-satoshi/issues/206 as well and we can create another issue once the logo has been finalized.

**Pull request closed without merging**